src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -123,6 +132,23 @@
         return fileInfos;
     }
 
+    /**
+     * Returns the list of non-binary files for the repo {@code repoConfig}.
+     */
+    public static HashSet<Path> getListOfNonBinaryFiles(RepoConfiguration repoConfig) {
+        String receivedMsg = GitDiff.gitGetModifiedFiles(Paths.get(repoConfig.getRepoRoot()));
+        String[] listOfFiles = receivedMsg.split("\n");
+
+        HashSet<String> nonBinaryFiles =  Arrays.stream(listOfFiles)
+                .filter(file -> !file.startsWith(BINARY_FILE_LINE_DIFF_RESULT))
+                .map(rawNonBinaryFile -> rawNonBinaryFile.split("\t")[2])
+                .collect(Collectors.toCollection(HashSet::new));
+
+        return nonBinaryFiles.stream().filter(fileName -> FileUtil.isValidPath(fileName))
```

Comment:
Why you do an unnecessary `collect` and `stream`? Can concatenate them together. If you are afraid of clearness, you can add comments.